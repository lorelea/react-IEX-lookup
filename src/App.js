import React from 'react';
import logo from './logo.svg';
import './App.scss';
import IEXInfoLookup from './IEXInfoLookup'

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
      </header>
      <main className="App-main">
        <div className="App-component">
          <IEXInfoLookup />
        </div>
      </main>
      <footer className="App-footer">
        <a
          className="App-link"
          href="https://iexcloud.io"
          target="_blank"
          rel="noopener noreferrer"
        >
          Data provided by IEX Cloud
        </a>
      </footer>
    </div>
  );
}

export default App;
