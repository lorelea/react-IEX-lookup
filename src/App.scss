@import 'styles/_colors.scss';
@import 'styles/scrollbar.scss';

.App {
  display: flex;
  flex-direction: column;
  text-align: center;
  height: 100vh;
  overflow: hidden;
  background-color: $app-bg;
  color: $app-text;

  &-logo {
    animation: App-logo-spin infinite 20s linear;
    height: 10vmin;
    pointer-events: none;
  }
  &-header {
    background-color: $app-header-bg;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    font-size: calc(10px + 2vmin);
    color: $app-header-text;
  }
  &-main {
    flex: 1;
    overflow: auto;
    padding: 0 1rem;
  }
  &-footer {
    padding: 0.5rem;
  }
  &-link {
    color: $app-link-text;
    
    &:hover {
      color: darken($app-link-text, 30);
    }
  }
  &-component {
    margin: 2rem auto 0 auto;
    max-width: 600px;
  }
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
