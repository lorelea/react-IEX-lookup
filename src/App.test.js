import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';

import { shallow } from 'enzyme';

it('App :: shallow :: renders without crashing', () => {
  shallow(<App />);
});

it('App :: renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("APP :: snapshot :: renders correctly", () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});