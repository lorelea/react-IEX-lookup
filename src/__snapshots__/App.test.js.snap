// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP :: snapshot :: renders correctly 1`] = `ShallowWrapper {}`;

exports[`renders correctly 1`] = `ShallowWrapper {}`;
